public/app/plugins/datasource/cloudwatch/datasource.ts

```diff
@@ -537,37 +555,47 @@ export class CloudWatchDatasource
     const requestParams = {
       from: range.from.valueOf().toString(),
       to: range.to.valueOf().toString(),
-      queries: queryParams.map((param: any) => ({
-        refId: 'A',
-        intervalMs: 1, // dummy
-        maxDataPoints: 1, // dummy
-        datasource: this.getRef(),
-        type: 'logAction',
-        subtype: subtype,
-        ...param,
-      })),
+      queries: queryParams.map(
+        (param: GetLogEventsRequest | StartQueryRequest | DescribeLogGroupsRequest | GetLogGroupFieldsRequest) => ({
+          refId: (param as StartQueryRequest).refId || 'A',
+          intervalMs: 1, // dummy
+          maxDataPoints: 1, // dummy
+          datasource: this.getRef(),
+          type: 'logAction',
+          subtype: subtype,
+          ...param,
+        })
+      ),
     };
 
     if (options.makeReplacements) {
-      requestParams.queries.forEach((query) => {
-        const fieldsToReplace: Array<
-          keyof (GetLogEventsRequest & StartQueryRequest & DescribeLogGroupsRequest & GetLogGroupFieldsRequest)
-        > = ['queryString', 'logGroupNames', 'logGroupName', 'logGroupNamePrefix'];
-
-        for (const fieldName of fieldsToReplace) {
-          if (query.hasOwnProperty(fieldName)) {
-            if (Array.isArray(query[fieldName])) {
-              query[fieldName] = query[fieldName].map((val: string) =>
-                this.replace(val, options.scopedVars, true, fieldName)
-              );
-            } else {
-              query[fieldName] = this.replace(query[fieldName], options.scopedVars, true, fieldName);
+      requestParams.queries.forEach(
```

Comment:
Not sure I understand this section here below. What is makeReplacements? Shouldn't template variables be interpolated [here](https://github.com/grafana/grafana/blob/aocenas/cw-logs-limit-retries/public/app/plugins/datasource/cloudwatch/datasource.ts#L162-L167) instead?

Reply:
To be honest not sure. Only thing I wanted to do here is to add correct refId here https://github.com/grafana/grafana/pull/39290/files/e5e0ae1f9130208fdfc94b3d56021846386da153#diff-5195d59bff5abb37dceeb2ba37e2e0acb2f73f861fc83e320a72f6e7b2784f18R560 and I added a correct type as an improvement. That resulted in some additional typing later. Didn't want to really change much else here.